### PR TITLE
React-capstone

### PR DESCRIPTION
**Pull request by rbd3**
### Change made:
👉 Use React components.
👉 Use React Router.
👉 Connect React and Redux.
👉 Write integration tests with a React testing library.
👉 Use styles in a React app.
👉 Use store, actions, and reducers in React.

### CODE REVIEW

File: src/styles/Home.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  justify-content: space-between;
+  align-items: center;
+  padding: 0 10px;
+  background-color: rgb(243, 204, 211);
+}
+
+.movies h1 {
+  font-family: Lato, sans-serif;
+  font-weight: 700;
+  font-size: 5rem;
```
Comment (zieeco) [inline src/styles/Home.css]: - Please, consider reducing the size, it's too big. this will improve your design.

File: src/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <Movie
+              key={movie.id}
+              Movie={movie}
+            />
+          ))}
+        </div>
+      </div>
+    </>
+  );
+}
```
Comment (zieeco) [inline src/components/Home.jsx]: - Hey, according to the project requirement, one should be able to search any movies by name. You've not implemented that. imagine your movie list has many movies, should users have to scroll down the whole page to view that movie? They should be able to search by the name, and the array of movies with the filter to return the searched movie. Please, implement this missing part of the project requirement to improve the user experience of your app.

*requirement*
image


*see an example of where you can add the search functionality*
image

File: src/tests/Home.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { MemoryRouter } from 'react-router-dom';
+/* eslint-disable */
```
Comment (zieeco) [inline src/tests/Home.test.js]: - Kindly remove this line of disabling the linter and fix it to improve your development skill, as it's regarded a bad practice and not in accordance with microverse to disable the linter.

File: src/tests/Home.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { toBeInTheDocument } from '@testing-library/jest-dom/extend-expect';
```
Comment (zieeco) [inline src/tests/Home.test.js]: - This is where you have the linter error, you've imported what you didn't use. kindly remove it.

File: src/tests/NavBar.test.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { render } from '@testing-library/react';
+import NavBar from '../components/NavBar';
+
+test('renders NavBar component', () => {
+  const { container } = render(<NavBar />);
+  expect(container.innerHTML).toMatchSnapshot();
+});
```
Comment (zieeco) [inline src/tests/NavBar.test.js]: - Your test is failing here. Please, double-check to fix this.

image

File: README.md [hunk 6]
```diff
@@ -1,70 +1,180 @@
...
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+<!-- LIVE DEMO -->
+## 🚀 Live Demo <a name="live-demo"></a>
 
-### Code Splitting
+- [Video Walkthrough](https://www.loom.com/share/3f2e5cf1460e460c9a15ecd81cca62a0?sid=343f1613-2e7b-4dfa-9257-d551870ad19e)
```
Comment (zieeco) [inline README.md]: - Great adding the presentation video, you should also add the live demo version as you should deploy the application using Render, Netlify, or GitHub Pages, and add the link to the deployed project in the README.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @rbd3,





Great job adding the associations 👏 
You've done well,  however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design ✔️
- Descriptive PR. ✔️
- Nice project implementation. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @rbd3, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**